src/docs/cookbook/navigation/named-routes-drawer.md

```diff
@@ -0,0 +1,236 @@
+---
+title: Navigate with named routes within a drawer
+---
+
+In the
+[Navigate with named routes](/docs/cookbook/navigation/named-routes/)
+recipe, we learned how to Navigate to the same screen in many parts of our apps
+via a "named route". In the
+[Add a Drawer to a screen](/docs/cookbook/design/drawer/) recipe, we
+learned how to user a drawer for Navigation.
+
+This example will replicate the functionality from the original two recipes,
+demonstrating how to combine named routes with a drawer to Navigate between
+pages at the root of an application.
+
+## Directions
+
+  1. Create a Drawer
+  1. Create two screens
+  3. Define the routes
+
+## 1. Create a Drawer
+
+In order to share Navigation logic used in the drawer between screens we'll
+create a `NavigationDrawer` widget. Within the `onTap` of our `ListTitle`
+widgets we'll use the
+[`Navigator.pushReplacementNamed`](https://docs.flutter.io/flutter/widgets/Navigator/pushReplacementNamed.html)
+function. This associates the list items in our `Drawer` with the screens we'll
+define our `routes` table.
+
+<!-- skip -->
+```dart
+class NavigationDrawer extends StatelessWidget  {
```

Comment:
Just to ensure we're creating the same NavigationDrawer every time, it might be good to provide a `const` Constructor here. That way you can always instantiate this with `const NavigationDrawer()`, which will ensure the same copy will always be re-used.

Otherwise, it might be good to accept a `key` here, and pass the same key on each page. If you were to provide a different key to each page, or if the structure of the pages was a bit different, Flutter might think these are two different things and create a new instance. Using a consistent `key` or `const` constructor should protect against that!